<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A Complete Deconvolution Method using Sequencing Data • CDSeq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="A Complete Deconvolution Method using Sequencing Data">
<meta property="og:description" content="Estimate cell-type-specific gene expression profiles and sample-specific cell-type proportions simultaneously using bulk sequencing data. Kang et al. (2019) &lt;doi:10.1371/journal.pcbi.1007510&gt;. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CDSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/CDSeq-vignette.html">CDSeq: a complete deconvolution method using sequencing data</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kkang7/CDSeq_R_Package/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="cdseq">CDSeq<a class="anchor" aria-label="anchor" href="#cdseq"></a>
</h1></div>
<!-- badges: start -->
<p>CDSeq is a complete deconvolution method for dissecting bulk RNA-Seq data. The input of CDSeq is, ideally, bulk RNA-Seq read counts (similar to the input format required by DESeq2), and CDSeq will estimate, simultaneously, the cell-type-specific gene expression profiles and the sample-specific cell-type proportions, no reference of pure cell line GEPs or scRNAseq reference is needed for running CDSeq.</p>
<p>For example, if you have a bulk RNA-Seq data, a G by M matrix <strong>A</strong>, which is a G by M matrix. G denotes the number of genes and M is the sample size, then CDSeq will output <strong>B</strong> (a G by T matrix) and <strong>C</strong> (a T by M matrix), where T is the number of cell types, <strong>B</strong> is the estimate of cell-type-specific GEPs and <strong>C</strong> is the estimate of sample-specific cell-type proportions.</p>
<p>Importantly, you can ask CDSeq to estimate the number of cell types, i.e. T, by providing a vector of possible integer values for T. For example, if the user input for T is a vector, i.e. (T={2,3,4,5,6}), then CDSeq will estimate the most likely number for T.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the released version of CDSeq from <a href="https://CRAN.R-project.org" class="external-link">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"CDSeq"</span><span class="op">)</span></span></code></pre></div>
<p>And the development version from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"kkang7/CDSeq_R_Package"</span><span class="op">)</span></span></code></pre></div>
<p>build the vignette with</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"kkang7/CDSeq_R_Package"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="known-issue-about-macos-installation">Known issue about MacOS installation<a class="anchor" aria-label="anchor" href="#known-issue-about-macos-installation"></a>
</h2>
<p>It is possible for Mac users to run into some errors when install from source due to problems of Rcpp compiler tools. Follow the instruction here may help: <a href="https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/" class="external-link uri">https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/</a></p>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>Load package</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/kkang7/CDSeq_R_Package" class="external-link">CDSeq</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="when-the-number-of-cell-types-is-a-scalar">When the number of cell types is a scalar<a class="anchor" aria-label="anchor" href="#when-the-number-of-cell-types-is-a-scalar"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## basic example code</span></span>
<span><span class="va">result1</span><span class="op">&lt;-</span><span class="fu"><a href="reference/CDSeq.html">CDSeq</a></span><span class="op">(</span>bulk_data <span class="op">=</span>  <span class="va">mixtureGEP</span>, </span>
<span>               cell_type_number <span class="op">=</span> <span class="fl">6</span>, </span>
<span>               mcmc_iterations <span class="op">=</span> <span class="fl">5</span>, <span class="co"># increase the mcmc_iterations to 700 or above</span></span>
<span>               gene_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">gene_length</span><span class="op">)</span>, </span>
<span>               reference_gep <span class="op">=</span> <span class="va">refGEP</span>,  <span class="co"># gene expression profile of pure cell lines</span></span>
<span>               cpu_number <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="when-the-number-of-cell-types-is-a-vector">When the number of cell types is a vector<a class="anchor" aria-label="anchor" href="#when-the-number-of-cell-types-is-a-vector"></a>
</h3>
<p>The <em>cell_type_number</em> can also be a vector which contains different integer values. CDSeq will perform estimation for each integer in the vector and estimate the number of cell types in the mixtures. For example, one can set <em>cell_type_number = 2:10</em> as follows, and CDSeq will estimate the most likely number of cell types from 2 to 10.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result2</span><span class="op">&lt;-</span><span class="fu"><a href="reference/CDSeq.html">CDSeq</a></span><span class="op">(</span>bulk_data <span class="op">=</span>  <span class="va">mixtureGEP</span>, </span>
<span>              cell_type_number <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>              mcmc_iterations <span class="op">=</span> <span class="fl">5</span>, </span>
<span>              dilution_factor <span class="op">=</span> <span class="fl">1</span>, </span>
<span>              block_number <span class="op">=</span> <span class="fl">1</span>, </span>
<span>              gene_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">gene_length</span><span class="op">)</span>, </span>
<span>              reference_gep <span class="op">=</span> <span class="va">refGEP</span>, <span class="co"># gene expression profile of pure cell lines</span></span>
<span>              cpu_number <span class="op">=</span> <span class="fl">1</span>, <span class="co"># use multiple cores to save time. Set the cpu_number = length(cell_type_number) if there is enough cores.</span></span>
<span>              print_progress_msg_to_file <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="use-single-cell-to-annotate-cdseq-estimated-cell-types">Use single cell to annotate CDSeq-estimated cell types<a class="anchor" aria-label="anchor" href="#use-single-cell-to-annotate-cdseq-estimated-cell-types"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cdseq.result</span> <span class="op">&lt;-</span> <span class="fu">CDSeq</span><span class="fu">::</span><span class="fu"><a href="reference/CDSeq.html">CDSeq</a></span><span class="op">(</span>bulk_data <span class="op">=</span> <span class="va">pbmc_mix</span>,</span>
<span>                             cell_type_number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">12</span>,<span class="fl">3</span><span class="op">)</span>,</span>
<span>                             beta <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                             alpha <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                             mcmc_iterations <span class="op">=</span> <span class="fl">700</span>,</span>
<span>                             cpu_number <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                             dilution_factor <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cdseq.result.celltypeassign</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cellTypeAssignSCRNA.html">cellTypeAssignSCRNA</a></span><span class="op">(</span>cdseq_gep <span class="op">=</span> <span class="va">cdseq.result</span><span class="op">$</span><span class="va">estGEP</span>, <span class="co"># CDSeq-estimated cell-type-specific GEPs</span></span>
<span>                                                   cdseq_prop <span class="op">=</span> <span class="va">cdseq.result</span><span class="op">$</span><span class="va">estProp</span>, <span class="co"># CDSeq-estimated cell type proportions</span></span>
<span>                                                   sc_gep <span class="op">=</span> <span class="va">sc_gep</span>,         <span class="co"># PBMC single cell data</span></span>
<span>                                                   sc_annotation <span class="op">=</span> <span class="va">sc_annotation</span>,<span class="co"># PBMC single data annotations</span></span>
<span>                                                   sc_pt_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                                   cdseq_pt_size <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                                                   seurat_nfeatures <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                                                   seurat_npcs <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                                                   seurat_dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,</span>
<span>                                                   plot_umap <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                                                   plot_tsne <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="setting-recommendations">Setting recommendations<a class="anchor" aria-label="anchor" href="#setting-recommendations"></a>
</h2>
<p>We provide recommendations for parameter settings. Note that these recommendations are merely emperical and there is no theoretical justifications yet. User can tune the parameters based on specific applications and domain knowledges.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Parameters</th>
<th align="left">Recommended setting</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">beta</td>
<td align="left">0.5</td>
</tr>
<tr class="even">
<td align="left">alpha</td>
<td align="left">5</td>
</tr>
<tr class="odd">
<td align="left">mcmc_iteration</td>
<td align="left">700-2000</td>
</tr>
<tr class="even">
<td align="left">dilution_factor</td>
<td align="left">2-10</td>
</tr>
<tr class="odd">
<td align="left">gene_subset_size</td>
<td align="left">200-500</td>
</tr>
<tr class="even">
<td align="left">block_number</td>
<td align="left">&gt;5</td>
</tr>
</tbody>
</table>
<p>Check vignette for more details and examples: <em>browseVignettes(“CDSeq”)</em>.</p>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>email: <a href="mailto:kangkai0714@gmail.com" class="email">kangkai0714@gmail.com</a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=CDSeq" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/kkang7/CDSeq_R_Package/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/kkang7/CDSeq_R_Package/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CDSeq</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Kai Kang <br><small class="roles"> Author, maintainer, copyright holder </small>  </li>
<li>Caizhi Huang <br><small class="roles"> Author </small>  </li>
<li>Leping Li <br><small class="roles"> Author, copyright holder </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.com/kkang7/CDSeq_R_Package" class="external-link"><img src="https://travis-ci.com/kkang7/CDSeq_R_Package.svg?branch=master" alt="Travis build status"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Kai Kang, Caizhi Huang, Leping Li.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
