<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Complete deconvolution using sequencing data. — CDSeq • CDSeq</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Complete deconvolution using sequencing data. — CDSeq"><meta property="og:description" content="CDSeq takes bulk RNA-seq data as input and simultaneously returns estimates of both cell-type-specific gene expression profiles and sample-specific cell-type proportions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CDSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/CDSeq-vignette.html">CDSeq: a complete deconvolution method using sequencing data</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/kkang7/CDSeq_R_Package/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Complete deconvolution using sequencing data.</h1>
    <small class="dont-index">Source: <a href="https://github.com/kkang7/CDSeq_R_Package/blob/HEAD/R/CDSeq.R" class="external-link"><code>R/CDSeq.R</code></a></small>
    <div class="hidden name"><code>CDSeq.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>CDSeq</code> takes bulk RNA-seq data as input and simultaneously returns estimates of both cell-type-specific gene expression profiles and sample-specific cell-type proportions.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">CDSeq</span><span class="op">(</span></span>
<span>  <span class="va">bulk_data</span>,</span>
<span>  beta <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  cell_type_number <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mcmc_iterations <span class="op">=</span> <span class="fl">700</span>,</span>
<span>  dilution_factor <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  gene_subset_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  block_number <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  cpu_number <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_length <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reference_gep <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  print_progress_msg_to_file <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>bulk_data</dt>
<dd><p>RNA-Seq read counts matrix. Columns represent samples and rows represent genes.</p></dd>


<dt>beta</dt>
<dd><p>beta is a scalar or a vector of length G where G is the number of genes; default value for beta is 0.5; When beta=Null, CDSeq uses reference_gep to estimate beta.</p></dd>


<dt>alpha</dt>
<dd><p>alpha is a scalar or a vector of length cell_type_number where cell_type_number is the number of cell type; default value for alpha is 5.</p></dd>


<dt>cell_type_number</dt>
<dd><p>number of cell types. cell_type_number can be an integer or a vector of different integers. To estimate the number of cell types, please provide a vector for cell_type_number,
e.g. cell_type_number &lt;- 2:30, then CDSeq will estimate the number of cell types.</p></dd>


<dt>mcmc_iterations</dt>
<dd><p>number of iterations for the Gibbs sampler; default value is 700.</p></dd>


<dt>dilution_factor</dt>
<dd><p>a scalar to dilute the read counts for speeding up; default value is 1. CDSeq will use bulk_data/dilution_factor.</p></dd>


<dt>gene_subset_size</dt>
<dd><p>number of genes randomly sampled for each block. Default is NULL.</p></dd>


<dt>block_number</dt>
<dd><p>number of blocks. Each block contains gene_subset_size genes. Default is 1.</p></dd>


<dt>cpu_number</dt>
<dd><p>number of cpu cores that can be used for parallel computing; 
Default is NULL and CDSeq will detect the available number of cores on the device and use number of all cores - 1 for parallel computing.</p></dd>


<dt>gene_length</dt>
<dd><p>a vector of the effective length (gene length - read length + 1) of each gene; Default is NULL.</p></dd>


<dt>reference_gep</dt>
<dd><p>a reference gene expression profile can be used to determine the cell type and/or estimate beta; Default is NULL.</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, then print progress message to the console. Default is FALSE.</p></dd>


<dt>print_progress_msg_to_file</dt>
<dd><p>print progress message to a text file. Set 1 if need to print progress msg to a file and set 0 if no printing. Default is 0;</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>CDSeq returns estimates of both cell-type-specific gene expression profiles and sample-specific cell-type proportions. CDSeq will also return estimated number of cell types.
 and the log posterior values for different number of cell types.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">result1</span><span class="op">&lt;-</span><span class="fu">CDSeq</span><span class="op">(</span>bulk_data <span class="op">=</span>  <span class="va">mixtureGEP</span>, cell_type_number <span class="op">=</span> <span class="fl">6</span>, mcmc_iterations <span class="op">=</span> <span class="fl">5</span>, </span></span>
<span class="r-in"><span>       dilution_factor <span class="op">=</span> <span class="fl">50</span>, block_number <span class="op">=</span> <span class="fl">1</span>, gene_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">gene_length</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>       reference_gep <span class="op">=</span> <span class="va">refGEP</span>, cpu_number<span class="op">=</span><span class="fl">1</span>, print_progress_msg_to_file<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CDSeq is running in non Reduce-Recover mode. To use Reduce-Recover mode, assign a value to block_number that is greater than 1.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CDSeq completed successfully using 0.0001 hours</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Kai Kang, Caizhi Huang, Leping Li.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

