<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>cellTypeAssignSCRNA assigns CDSeq-identified cell types using single cell RNAseq data. — cellTypeAssignSCRNA • CDSeq</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="cellTypeAssignSCRNA assigns CDSeq-identified cell types using single cell RNAseq data. — cellTypeAssignSCRNA"><meta property="og:description" content="cellTypeAssignSCRNA assigns CDSeq-identified cell types using single cell RNAseq data."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CDSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/CDSeq-vignette.html">CDSeq: a complete deconvolution method using sequencing data</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/kkang7/CDSeq_R_Package/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1><code>cellTypeAssignSCRNA</code> assigns CDSeq-identified cell types using single cell RNAseq data.</h1>
    <small class="dont-index">Source: <a href="https://github.com/kkang7/CDSeq_R_Package/blob/HEAD/R/cellTypeAssignSCRNA.R" class="external-link"><code>R/cellTypeAssignSCRNA.R</code></a></small>
    <div class="hidden name"><code>cellTypeAssignSCRNA.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>cellTypeAssignSCRNA</code> assigns CDSeq-identified cell types using single cell RNAseq data.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">cellTypeAssignSCRNA</span><span class="op">(</span></span>
<span>  cdseq_gep <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cdseq_prop <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cdseq_gep_sample_specific <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sc_gep <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sc_annotation <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sc_batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  batch_correction <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  harmony_iter <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  harmony_cluster <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  nb_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nb_mu <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  corr_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  breaksList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.01</span><span class="op">)</span>,</span>
<span>  pseudo_cell_count <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  seurat_count_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  seurat_scale_factor <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  seurat_norm_method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>  seurat_select_method <span class="op">=</span> <span class="st">"vst"</span>,</span>
<span>  seurat_nfeatures <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  seurat_npcs <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  seurat_dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,</span>
<span>  seurat_reduction <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>  seurat_resolution <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  seurat_find_marker <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  seurat_DE_test <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  seurat_DE_logfc <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  seurat_top_n_markers <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  sc_pt_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  cdseq_pt_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  plot_umap <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  plot_tsne <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  plot_per_sample <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  fig_save <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  fig_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  fig_name <span class="op">=</span> <span class="st">"CDSeqCellTypeAssignSCRNA"</span>,</span>
<span>  fig_format <span class="op">=</span> <span class="st">"jpeg"</span>,</span>
<span>  fig_dpi <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  corr_heatmap_fontsize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>cdseq_gep</dt>
<dd><p>CDSeq-estimated gene expression profile matrix with G rows (genes) and T columns (cell types).</p></dd>


<dt>cdseq_prop</dt>
<dd><p>CDSeq-estimated sample-specific cell-type proportion, a matrix with T rows (cell type) and M (sample size).</p></dd>


<dt>cdseq_gep_sample_specific</dt>
<dd><p>CDSeq-estimated sample-specific cell type gene expression, in the form of read counts. It is a 3 dimension array, i.e. gene by sample by cell type. The element cdseq_gep_sample_specific[i,j,k] represents the reads mapped to gene i from cell type k in sample j.</p></dd>


<dt>sc_gep</dt>
<dd><p>a G (genes) by N (cell) matrix or dataframe that contains the gene expression profile for N single cells.</p></dd>


<dt>sc_annotation</dt>
<dd><p>a dataframe contains two columns "cell_id"  and "cell_type". cell_id needs to match with the cell_id in sc_gep but not required to have the same size. cell_type is the cell type annotation for the single cells.</p></dd>


<dt>sc_batch</dt>
<dd><p>a vector contains batch information of single cell data, i.e. sc_gep, and length(sc_batch) = ncol(sc_gep).</p></dd>


<dt>batch_correction</dt>
<dd><p>perform Harmony batch correction if it is 1.</p></dd>


<dt>harmony_iter</dt>
<dd><p>Maximum number of rounds to run Harmony. One round of Harmony involves one clustering and one correction step.</p></dd>


<dt>harmony_cluster</dt>
<dd><p>Maximum number of rounds to run clustering at each round of Harmony.</p></dd>


<dt>nb_size</dt>
<dd><p>size parameter for negative binomial distribution, check rnbinom for details.</p></dd>


<dt>nb_mu</dt>
<dd><p>mu parameter for negative binomial distribution, check rnbinom for details.</p></dd>


<dt>corr_threshold</dt>
<dd><p>if the correlation between CDSeq-estimated GEPs and the scRNAseq GEP is below this value, then it is considered the two cell types are not matching.</p></dd>


<dt>breaksList</dt>
<dd><p>parameter for pheatmap controling the color scale. See pheatmap function for details.</p></dd>


<dt>pseudo_cell_count</dt>
<dd><p>an integer indicating how many pseudo cells will be generated from CDSeq-estimated cell-type-specific gene expression profiles. Default values is 1.</p></dd>


<dt>seurat_count_threshold</dt>
<dd><p>this parameter will be passed to Seurat subset function (subset = nCount_RNA &gt; seurat_count_threshold) for filtering out single cells whose total counts is less this threshold.</p></dd>


<dt>seurat_scale_factor</dt>
<dd><p>this parameter will be passed to scale.factor in Seurat function NormalizeData.</p></dd>


<dt>seurat_norm_method</dt>
<dd><p>this parameter will be passed to normalization.method in Seurat function NormalizeData.</p></dd>


<dt>seurat_select_method</dt>
<dd><p>this parameter will be passed to selection.method in Seurat function FindVariableFeatures</p></dd>


<dt>seurat_nfeatures</dt>
<dd><p>this parameter will be passed to nfeatures in Seurat function FindVariableFeatures.</p></dd>


<dt>seurat_npcs</dt>
<dd><p>this parameter will be passed to npcs in Seurat function RunPCA.</p></dd>


<dt>seurat_dims</dt>
<dd><p>this parameter will be passed to dims in Seurat function FindNeighbors.</p></dd>


<dt>seurat_reduction</dt>
<dd><p>this parameter will be passed to reduction in Seurat function FindNeighbors.</p></dd>


<dt>seurat_resolution</dt>
<dd><p>this parameter will be passed to resolution in Seurat function FindClusters.</p></dd>


<dt>seurat_find_marker</dt>
<dd><p>this parameter controls if run seurat FindMarker function, default is FALSE.</p></dd>


<dt>seurat_DE_test</dt>
<dd><p>this parameter will be passed to test.use in Seurat function FindAllMarkers.</p></dd>


<dt>seurat_DE_logfc</dt>
<dd><p>this parameter will be passed to logfc.threshold in Seurat function FindAllMarkers.</p></dd>


<dt>seurat_top_n_markers</dt>
<dd><p>the number of top DE markers saved from Seurat output.</p></dd>


<dt>sc_pt_size</dt>
<dd><p>point size of single cell data in umap and tsne plots</p></dd>


<dt>cdseq_pt_size</dt>
<dd><p>point size of CDSeq-estimated cell types in umap and tsne plots</p></dd>


<dt>plot_umap</dt>
<dd><p>set 1 to plot umap figure of scRNAseq and CDSeq-estimated cell types, 0 otherwise.</p></dd>


<dt>plot_tsne</dt>
<dd><p>set 1 to plot tsne figure of scRNAseq and CDSeq-estimated cell types, 0 otherwise.</p></dd>


<dt>plot_per_sample</dt>
<dd><p>currently disabled for debugging</p></dd>


<dt>fig_save</dt>
<dd><p>1 or 0. 1 means save figures to local and 0 means do not save figures to local.</p></dd>


<dt>fig_path</dt>
<dd><p>the location where the heatmap figure is saved.</p></dd>


<dt>fig_name</dt>
<dd><p>the name of umap and tsne figures. Umap figure will have the name of fig_name_umap_date and tsne figure will be named fig_name_tsne_date.</p></dd>


<dt>fig_format</dt>
<dd><p>"pdf", "jpeg", or "png".</p></dd>


<dt>fig_dpi</dt>
<dd><p>figure dpi</p></dd>


<dt>corr_heatmap_fontsize</dt>
<dd><p>font size of the correlation heatmap between scRNAseq GEP and CDSeq-estimated GEPs.</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, some calculation information will be print.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>cellTypeAssignSCRNA returns a list containing following fields: 
fig_path: same as the input fig_path</p>


<p>fig_name: same as the input fig_name</p>


<p>cdseq_synth_scRNA: synthetic scRNAseq data generated using CDSeq-estiamted GEPs</p>


<p>cdseq_scRNA_umap: ggplot figure of the umap outcome</p>


<p>cdseq_scRNA_tsne: ggplot figure of the tsne outcome</p>


<p>cdseq_synth_scRNA_seurat: Seurat object containing the scRNAseq combined with CDSeq-estimated cell types. Cell id for CDSeq-estimated cell types start with "CDSeq".</p>


<p>seurat_cluster_purity: for all cells in a Seurat cluster i,  the ith value in seurat_cluster_purity is the proportion of the mostly repeated cell annotation from sc_annotation. 
For example, after Seurat clustering, suppose there are 100 cells in cluster 1, out of these 100 cells, 90 cells' annotation in sc_annotation is cell type A, then 
the fist value in seurat_cluster_purity is 0.9. This output can be used to assess the agreement between Seurat clustering and the given sc_annotation.</p>


<p>seurat_unique_clusters: Unique Seurat cluster numbering. This can be used together with seurat_cluster_gold_label to match the Seurat clusters with given annotations.</p>


<p>seurat_cluster_gold_label: The cell type annotations for each unique Seurat cluster based on sc_annotation.</p>


<p>seurat_markers: DE genes for each Seurat cluster.</p>


<p>seurat_top_markers: Top seurat_top_n_markers DE genes for each Seurat cluster.</p>


<p>CDSeq_cell_type_assignment_df: cell type assignment for CDSeq-estimated cell types.</p>


<p>CDSeq_cell_type_assignment_confidence: cell type assignment confidence matrix, only available when pseudo_cell_count &gt; 1.</p>


<p>CDSeq_cell_type_assignment_df_all: cell type assignment for CDSeq-estimated cell types, only available when pseudo_cell_count &gt; 1.</p>


<p>cdseq_prop_merged: CDSeq-estimated cell type proportions with cell type annotations (annotated using clustering with scRNAseq).</p>


<p>cdseq_gep_sample_specific_merged: sample-specific cell-type read counts. It is a 3d array with dimensions: gene, sample, cell type.</p>


<p>input_list: values for input parameters</p>


<p>cdseq_sc_comb_umap_df: dataframe for umap plot</p>


<p>cdseq_sc_comb_tsne_df: dataframe for tsne plot</p>


<p>cdseq_prop_merged_byCorr: CDSeq-estimated cell type proportions with cell type annotations (annotated using correlation with scRNAseq).</p>


<p>cdseq_gep_merged_byCorr: CDSeq-estimated cell-type-specific GEPs with cell type annotations (annotated using correlation with scRNAseq).</p>


<p>cdseq_annotation_byCorr: CDSeq-estimated cell type annotations (annotated using correlation with scRNAseq)</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Kai Kang, Caizhi Huang, Leping Li.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

